imsim/skycat.py

```diff
@@ -0,0 +1,374 @@
+"""
+Interface to obtain objects from skyCatalogs.
+"""
+import math
+import numpy as np
+from galsim.config import InputLoader, RegisterInputType, RegisterValueType, \
+    RegisterObjectType, RegisterSEDType
+from galsim import CelestialCoord
+import galsim
+from desc.skycatalogs import skyCatalogs
+from .instcat import get_radec_limits
+
+
+class SkyCatalogInterface:
+    """Interface to skyCatalogs package."""
+    _bp500 = galsim.Bandpass(galsim.LookupTable([499,500,501],[0,1,0]),
+                             wave_type='nm').withZeropoint('AB')
+
+    # Using area-weighted effective aperture over FOV
+    # from https://confluence.lsstcorp.org/display/LKB/LSST+Key+Numbers
+    _rubin_area = 0.25 * np.pi * 649**2  # cm^2
+
+    def __init__(self, file_name, wcs, obj_types=None, edge_pix=100,
+                 flip_g2=True, logger=None):
+        """
+        Parameters
+        ----------
+        file_name : str
+            Name of skyCatalogs yaml config file.
+        wcs : galsim.WCS
+            WCS of the image to render.
+        obj_types : list-like [None]
+            List or tuple of object types to render, e.g., ('star', 'galaxy').
+            If None, then consider all object types.
+        edge_pix : float [100]
+            Size in pixels of the buffer region around nominal image
+            to consider objects.
+        flip_g2 : bool [True]
+            Flag to flip sign of g2.
```

Comment:
Please tell me we don't still have the g2 values flipped!  I would hope that this should default to False, and even better that we could remove it entirely here.

Reply:
I would like to remove it, but it seems to be in both the shear convention and in the convention for the PA of the galaxy.  So I kept it in here for now since I wanted to compare directly to runs using instance catalogs so I could have the flexibility to toggle it and then figure out where it can be removed.

Reply:
OK.  Hopefully we can fix the convention upstream in the skycats themselves and remove it here.